test/e2e/magic_catalog.go

```diff
@@ -149,7 +131,38 @@ func (c *magicCatalog) UndeployCatalog(ctx context.Context) []error {
 	return c.undeployCatalog(ctx, resourcesInOrderOfDeletion)
 }
 
-func (c *magicCatalog) undeployCatalog(ctx context.Context, resources []k8scontrollerclient.Object) []error {
+func (c *MagicCatalog) catalogSourceIsReady(ctx context.Context) error {
+	// wait for catalog source to become ready
+	key := k8scontrollerclient.ObjectKey{
+		Name:      c.name,
+		Namespace: c.namespace,
+	}
+
+	return waitFor(func() (bool, error) {
```

Comment:
If we're going to refactor, why not use Ginkgo for this like the rest of the suite does?

Reply:
ah, I see this is using ginkgo at some level.